src/Compilers/CSharp/Test/Emit2/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -17,6 +17,111 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests;
 
 public class PatternMatchingTests_ListPatterns : PatternMatchingTestBase
 {
+    // PROTOTYPE: Wrap entire dag in a try-finally for disposal, possibly one per each enumerator instance in case Dispose throws.
+    // PROTOTYPE: Compute buffer size based on forthcoming element evaluations
+    // PROTOTYPE: Subsumption checking is different for enumerable list-patterns because we cannot test the length upfront.
+    // PROTOTYPE: Instead, we would try to find alternative indexes and rewrite the input temps everywhere which will cause the values to merge.
+    // PROTOTYPE: It's possible we could get rid of double dag building and synthesized nodes only if we accept regression in Subsumption_03 and the like.
```

Comment:
@jcouv Is this something you'd consider acceptable? This is not a problem for enumerables because we can't directly match the count and we fetch elements one-by-one, so the only way to do it is to rewrite identical inputs from get-go. A similar approach can be used for indexables with some regression in rare cases, however, then we can avoid building a second dag altogether. (also if you have any feedback on the impl so far feel free to chime in, thanks.)